tests/Makefile

```diff
@@ -1,59 +1,73 @@
-TELEMETRY_ENDPOINT_INSECURE=false
+CONFORMANCE_PREFIX = conformance-test-runner## Prefix for the conformance test runner image
+CONFORMANCE_TAG = latest## Tag for the conformance test runner image
+GATEWAY_CLASS = nginx## Gateway class to use
+GINKGO_FLAGS =
+GINKGO_LABEL =
+GW_API_PREV_VERSION ?= 1.0.0## Supported Gateway API version from previous NGF release
+GW_SERVICE_TYPE=NodePort## Service type to use for the gateway
+GW_SVC_GKE_INTERNAL=false
+K8S_VERSION ?= latest## Kuberentes version to use. Expected format: 1.24 (major.minor) or latest
+NGF_VERSION ?= $(shell git describe --tags $(shell git rev-list --tags --max-count=1))## NGF version to be tested (defaults to latest tag)
+PULL_POLICY=Never## Pull policy for the images
+PROVISIONER_MANIFEST = conformance/provisioner/provisioner.yaml
+SUPPORTED_FEATURES = HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPRoutePortRedirect,HTTPRouteSchemeRedirect,HTTPRouteHostRewrite,HTTPRoutePathRewrite,GatewayPort8080,HTTPRouteResponseHeaderModification,GRPCExactMethodMatching,GRPCRouteListenerHostnameMatching,GRPCRouteHeaderMatching
 
 ifneq ($(GINKGO_LABEL),)
     override GINKGO_FLAGS += --label-filter "$(GINKGO_LABEL)"
 endif
 
+.PHONY: update-go-modules
+update-go-modules: ## Update the gateway-api go modules to latest main version
+	go get -u sigs.k8s.io/gateway-api@main
+	go mod tidy
+
+.PHONY: build-test-runner-image
+build-test-runner-image: ## Build conformance test runner image
+	docker build -t $(CONFORMANCE_PREFIX):$(CONFORMANCE_TAG) -f conformance/Dockerfile .
+
+.PHONY: run-conformance-tests
+run-conformance-tests: ## Run conformance tests
+	kind load docker-image $(CONFORMANCE_PREFIX):$(CONFORMANCE_TAG)
+	kubectl apply -f conformance/conformance-rbac.yaml
+	kubectl run -i conformance \
+		--image=$(CONFORMANCE_PREFIX):$(CONFORMANCE_TAG) --image-pull-policy=Never \
+		--overrides='{ "spec": { "serviceAccountName": "conformance" }  }' \
+		--restart=Never -- sh -c "go test -v . -tags conformance,experimental -args --gateway-class=$(GATEWAY_CLASS) \
+						        --supported-features=$(SUPPORTED_FEATURES) --version=$(NGF_VERSION) \
+								--report-output=output.txt; cat output.txt" | tee output.txt
+	bash scripts/check-pod-exit-code.sh
+	sed -e '1,/CONFORMANCE PROFILE/d' output.txt > conformance-profile.yaml
+	rm output.txt
+	grpc_core_result=`yq '.profiles[0].core.result' conformance-profile.yaml`; \
+	http_core_result=`yq '.profiles[1].core.result' conformance-profile.yaml`; \
+	http_extended_result=`yq '.profiles[1].extended.result' conformance-profile.yaml`; \
+	if [ "$$grpc_core_result" != "failure" ] && [ "$$http_core_result" != "failure" ] && [ "$$http_extended_result" != "failure" ] ; then \
+		exit 0; \
+	else \
+		exit 2; \
+	fi
+
+.PHONY: cleanup-conformance-tests
+cleanup-conformance-tests: ## Clean up conformance tests fixtures
+	kubectl delete pod conformance
+	kubectl delete -f tests/conformance-rbac.yaml
+
+.PHONY: build
+build: generate-static-deployment
```

Comment:
this target will just add to the `build` target in the main Makefile, adding a requirement for the static files

Reply:
Oh, I didn't know it worked like that. How does it import the parent Makefile?

Reply:
with `-include ../Makefile`. It's in the middle of the file since we want some commands to come before the one in there, like this build that needs to generate the files before the binary is built.